### PR TITLE
Expanded the reach of value classes.

### PR DESCRIPTION
**Pull request by paulp**
Now extending AnyVal:
- RichInt, RichDouble, etc.
- ArrayOps.ofRef, ofBoolean, etc
- StringAdd
- StringFormat

The rest of it is the changes necessary to enable those.

### CODE REVIEW

**Comment by paulp** [comment]
I opened this against 2.10.x just in case it is desired; I will let someone else decide that.  Let me know and I'll flip it to master.  In the meantime, review by @jsuereth .

File: src/library/scala/collection/mutable/IndexedSeqLike.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ import generic._
  *  @define willNotTerminateInf
  *  @define mayNotTerminateInf
  */
-trait IndexedSeqLike[A, +Repr] extends scala.collection.IndexedSeqLike[A, Repr] { self =>
+trait IndexedSeqLike[A, +Repr] extends Any with scala.collection.IndexedSeqLike[A, Repr] { self =>
```
Comment (jsuereth) [inline src/library/scala/collection/mutable/IndexedSeqLike.scala:39]: This just blew my mind.   No issues showed up in tests?

**Comment by jsuereth** [comment]
This really does LGTM.  I'd live to see @gkossakowski performance numbers on this early if we can.   If all tests pass, I'd love to just press that green button that's calling my name, but let's be a bit cautious first.  I'll feed it through the community build and we'll see what @gkossakowski reports.

File: src/library/scala/collection/mutable/IndexedSeqLike.scala (hunk 1 above)
Comment (paulp) [inline src/library/scala/collection/mutable/IndexedSeqLike.scala:39]: Tested like a champ.  But see my email to scala-internals.

**Comment by paulp** [comment]
I threw in a couple more; this was crashing the compiler, but I realized it was SI-6215 and worked around it.

**Comment by gkossakowski** [comment]
Hi guys. I'm overloaded with non-technical work I need to do and tomorrow I'm going on vacations for one week so no chance for benchmarking data. Sorry :-(

**Comment by jsuereth** [comment]
Epic win on community build.  Here's a few stats:

```
name           before     after
------------------------------------------
scalacheck   8744.6    8420.7
scalatest     34401.8  31531.2
scala-stm    14161.7  12011.2
```

that's w/ average of 10 compiles on each project.

I'm merging!  NO new community build issues.   I need to automate this communtiy build A:B comparison stuff though.

**Pull request merged**